src/metabase/query_processor/middleware/optimize_temporal_filters.clj

```diff
@@ -114,8 +114,17 @@
 (defmethod can-optimize-filter? :between
   [filter-clause]
   (lib.util.match/match-one filter-clause
-    [_
-     (field :guard optimizable-expr?)
+    [:between
```

Comment:
The PR comment explained this case nicely, but it would be great to put that context into a comment here. It's not obvious what's special about this pattern without knowing that.